src/components/benchmark/VisualizeBenchmarks.tsx

```diff
@@ -1,62 +1,117 @@
+    return colorScale(value);
   }
 
-  // Generate random benchmark values
-  const benchmarkValues: BenchmarkTable = {};
-  workflows.forEach((workflow: WorkflowSolution, index: number) => {
-    const n_proteins = Math.floor(Math.random() * 100);
-    const availability = Math.floor(Math.random() * 100);
-    const executedSteps = Math.floor(Math.random() * (workflow.workflow_length + 1));
-    benchmarkValues[workflow.descriptive_name] = {
-      '1': {description: "Sample desc 1", value: workflow.workflow_length, desirability_value: (workflow.workflow_length / 10.0)} as BenchmarkValue,
-      '2': {description: "Sample desc 2", value: executedSteps, desirability_value: (executedSteps) / workflow.workflow_length} as BenchmarkValue,
-      '3': {description: "Sample desc 3", value: n_proteins, desirability_value: 0.01 * n_proteins} as BenchmarkValue,
-      '4': {description: "Sample desc 4", value: availability, desirability_value: 0.01 * availability} as BenchmarkValue
-    };
-  });
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file) {
+      // Read and parse file
+      const reader = new FileReader();
+      reader.readAsText(file, "UTF-8");
+      reader.onload = (evt) => {
+        const content = evt.target?.result;
+        const obj = JSON.parse(content as string);
+        setBenchmarkValues(obj);
+      }
+    }
+  }
+
+  const handleExpand = (rowKey: string) => {
+    const newState = { ...expandedRows };
+    newState[rowKey] = !newState[rowKey];
+    setExpandedRows(newState);
+  };
+
+  const tableRow = (label: string, key: string, benchmarkValues: BenchmarkValue[], isWorkflow: boolean) => {
+    const isExpanded: boolean = expandedRows[key];
+    const bgColor = isWorkflow ? 'beige' : 'white';
+    return (<tr key={key}>
+      {/* Expand button */}
+      <td style={{padding: "8px", backgroundColor: bgColor}}>{isWorkflow ? 
+        (<button className='btn btn-primary btn-square btn-sm' onClick={() => handleExpand(key)}>{isExpanded ? '-' : '+'}</button>) : []}</td>
+
+      {/* Workflow / tool label */}
+      <td style={{padding: "8px", backgroundColor: bgColor}}>{label}</td>
+
+      {/* Benchmark values */}
+      { benchmarkValues.map((bmv: BenchmarkValue, index: number) => {
+        const color = mapValueToColor(bmv.desirability);
+        const tooltip = bmv.tooltip;
+        return (<td key={index} style={{textAlign: "center", padding: "8px", backgroundColor: bgColor}}>
+          <span style={{backgroundColor: color}} 
+                className={`benchmark-value ${tooltip ? 'tooltip' : ''}`}
+                {...(tooltip ? { 'data-tip': tooltip } : {})}>
+            {bmv.value.toString()}
+          </span>
+        </td>);
+      })}
+    </tr>);
+  }
 
   return (<div>
     <div className="m-20">
+
+      {/* Upload button for json file */}
```

Comment:
It would be nice if we could store the execution results (where the users can opt-in/out), to be able to evaluate the quality of the generated workflows later on.

Reply:
Not at the moment, but we could think about it in the future if we have time